_posts/2018-11-06-mongodb-transactions.md

```diff
@@ -0,0 +1,157 @@
+They are distinct entities, but there is a relationship between them in the form of ownership information.
+If you update the ownership information on the "customer" document, you probably need to update it on the "car" document (or documents) as well.
+
+The only way to do this with guaranteed consistency is through a multi-document transaction.
+
+### Event processing
+Another use case of transactions is event processing.
+When a certain event occurs, it may need to atomically create, update, delete several entities at the same time.
+
+The example that was given in Aly's presentation was the creation or invalidation of a customer's account, which would require an update to all of the customer's entities.
+
+### Event logging or auditing
+Consider the case where, for logging or auditing purposes, you want to create an event trail of all changes that happen to a certain document or collection and you want to store this event trail in another collection.
+
+The event trail should be representative of what really occurred, so events that never occurred should not be logged nor should events that actually happened be lost.
+
+The only way to achieve this is to put the update and the logging of the update inside the same transaction.
+
+## Technical details
+Now let's explore some of the more technical features of multi-document transactions.
+
+### Atomicity
+This is pretty straightforward.
+In MongoDB transactions are atomic, which means that execution of multiple changes inside a transaction is an all-or-nothing deal: either all updates get committed, or none.
+
+### Snapshot isolation
+When you start a transaction, MongoDB creates a snapshot of the current state of the database.
+During your transaction you will not see any updates made by other sessions.
+You are *isolated* from them.
+This guarantees that throughout the transaction your session will see one consistent version of the data.
+
+Internally MongoDB uses an update structure inside the WiredTiger cache to maintain this consistent view on the database.
+This structure grows as writes occur to the database and is only cleaned out of the database once the transaction is committed or aborted.
+The implication of this is that long running transactions or a high write volume can put pressure on the cache, so it's recommended to keep the duration of any transaction as low as possible.
+
+To minimize cache pressure, you can use the server parameter 'transactionLifeTimeLimitSeconds' to set a sensible maximum transaction time.
+If a transaction runs for a longer time than this value, it will be aborted.
+The default value is 60 seconds.
+
+A sidenote for those DBAs or developers who are already familiar with Oracle: the update structure in the WiredTiger cache is similar to how rollback segments and undo tablespaces work in Oracle, except that the snapshot information is kept in memory instead of on disk and you don't have to manage it.
```

Comment:
```suggestion
A sidenote for those DBAs or developers who are already familiar with Oracle.
The update structure in the WiredTiger cache is similar to how rollback segments and undo tablespaces work in Oracle.
Except that the snapshot information is kept in memory instead of on disk and you don't have to manage it yourself.
```

Reply:
Het dubbel punt staat op zijn plaats, een zin starten met 'except' is not done.